src/util/renaming_backend.rs

```diff
@@ -0,0 +1,84 @@
+use candle_core::{DType, Device, Shape, Tensor};
+use candle_nn::var_builder::SimpleBackend;
+use candle_nn::Init;
+
+/// A backend that renames tensors before passing them to the inner backend.
+pub struct RenamingBackend<B, F>
+where
+    B: SimpleBackend,
+    F: Fn(&str) -> String + Send + Sync,
+{
+    inner: B,
+    rename_to_inner: F,
+}
+
+impl<B, F> RenamingBackend<B, F>
+where
+    B: SimpleBackend,
+    F: Fn(&str) -> String + Send + Sync,
+{
+    /// Create a new renaming backend.
+    pub fn new(inner: B, rename_to_inner: F) -> Self {
+        Self {
+            inner,
+            rename_to_inner,
+        }
+    }
+}
+
+impl<B, F> SimpleBackend for RenamingBackend<B, F>
+where
+    B: SimpleBackend,
+    F: Fn(&str) -> String + Send + Sync,
+{
+    fn get(
+        &self,
+        s: Shape,
+        name: &str,
+        h: Init,
+        dtype: DType,
+        dev: &Device,
+    ) -> candle_core::Result<Tensor> {
+        self.inner
+            .get(s, &(self.rename_to_inner)(name), h, dtype, dev)
```

Comment:
Shouldn't `self.rename_to_inner` automatically be an immutable borrow here as it's a `&self` method?

Reply:
The result of the call must be borrowed, since `get` takes a `&str`.